### PR TITLE
Bytearray Fix

### PR DESCRIPTION
**Pull request by jtpereyda**
Issue #1 
- All CRC calculate methods now take `bytearray` parameters as well as `bytes`.
- Added unit tests to verify `bytearray` is accepted.
- Updated `.gitignore`. For me it blocked .py files, so I used `git add -f`, which also added all the `.pyc` files. Oops. Now it blocks only .pyc, .pyo, pyd.

All tests pass; test system is an Ubuntu 14.04 distro with Python 3.4.0 (AKA "my machine").

### CODE REVIEW

File: tests/test_CRC16.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ def testCalculate(self):
         msg = "Calculated CRC16 for 0123456789 should be 0x443D"
         self.assertEqual(self.crc.calculate("0123456789"), int('0x443D', 0), msg)
 
+    def testCalculateBytearray(self):
+        msg = "Calculated CRC16 for 0123456789 should still be 0x443D" \
+              " when passing a bytearray parameter."
+        self.assertEqual(self.crc.calculate(bytearray("0123456789".encode('utf-8'))), int('0x443D', 0), msg)
```
Comment (cristianav) [inline tests/test_CRC16.py]: please move self.crc.calculate to the next line, similar to the other (modified) files

**Comment by cristianav** [comment]
It's not mandatory, but it would be nice to add your name into Contributors section of AUTHORS.rst.
You did a very good job, thank you!

File: tests/test_CRC16.py (hunk 1 above)
Comment (jtpereyda) [inline tests/test_CRC16.py]: Is there a style guideline behind this? I don't recall why I put it on a separate line in the other files, but I don't think that sort of thing is prescribed by PEP8.

If it's the style for this code repo though, I can change that.

Comment (cristianav) [inline tests/test_CRC16.py]: PEP8 has some [ references](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) regarding max line length. There is no written (or different than PEP8) guideline behind it, but as far as I can recall, I tried to keep line length under 100 chars.

**Comment by jtpereyda** [comment]
You're very welcome! Fixes added, let me know if you see anything else.

- Joshua

**Pull request merged**